docs/sources/enterprise/query-caching.md

```diff
@@ -15,7 +15,7 @@ weight: 300
 
 When query caching is enabled, Grafana temporarily stores the results of data source queries. When you or another user submit the exact same query again, the results will come back from the cache instead of from the data source (like Splunk or ServiceNow) itself.
 
-Query caching works for all backend data sources, and queries sent through the data source proxy. You can enable the cache globally and configure the cache duration (also called Time to Live, or TTL).
```

Comment:
I don't see a reason that it wouldn't work for all backend data sources. The only data sources that send requests through the data source proxy are frontend data sources, right?

Reply:
But is there an easy way for the user to know if a data source is backend or frontend? If so, we should document it; if not, we shouldn't keep that sentence in the documentation IMO.

Reply:
The only way to tell without looking at the code is to see if the caching tab is there I think.

Reply:
sorry to jump in into this discussion, but "backend data sources, and queries sent through the data source proxy" is nearly everyhing. it's the same as "all data sources, except those that use access=browser", if i remember correctly.

(i know nothing about how query-caching works 😄  , i'm only talking about what data-source-proxy means here)

Reply:
Is it possible to have a backend datasource with "access=browser"? That doesn't sound right to me.

Reply:
i don't think it is possible. it should be like this:
- backend-datasource: the request goes browser -> datasource_specific_go_code -> database
- frontend-datasource:
    - access=server: the request goes browser -> generic_grafana_go_code -> database
    - access=browser: the request goes browser -> dateabase

Reply:
right, so to me the text here is still true. Caching only works on backend data sources. The original mistake was that the documentation said it was possible to cache requests through the data source proxy but that was actually wrong. We are not able to cache requests that go through the proxy.